qa/tasks/restart.py

```diff
@@ -48,14 +48,11 @@ def get_tests(ctx, config, role, remote, testdir):
             'mkdir', '--', srcdir,
             run.Raw('&&'),
             'git',
-            'archive',
-            '--remote=git://git.ceph.com/ceph.git',
-            '%s:qa/workunits' % refspec,
-            run.Raw('|'),
-            'tar',
-            '-C', srcdir,
-            '-x',
-            '-f-',
+            'clone',
+            'https://git.ceph.com/ceph.git',
+            '--branch', '%s:qa/workunits' % refspec,
```

Comment:
The `--branch` couldn't specify the `qa/workunits` here. It's just for branch. And it won't work if the `refspec` is a tag or commit.

Reply:
I think we have to split this into 3 steps:
1, clone the whole `ceph` repo
2, checkout the `commit/tag/branch`
3, then change directory to `qa/workunits/`.